sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -373,11 +376,24 @@ def get_cache_manager(self, pipeline, create_if_absent=False):
               '/'.join(cache_root_path.parts[1:]), id(pipeline))
         else:
           cache_dir = tempfile.mkdtemp(dir=cache_root)
+          if not isinstance(pipeline_runner, direct_runner.DirectRunner):
+            _LOGGER.warning(
+                'A local cache directory has been specified while '
+                'not using DirectRunner. It is recommended to cache into a '
+                'GCS bucket instead.')
       else:
-        cache_dir = tempfile.mkdtemp(
-            suffix=str(id(pipeline)),
-            prefix='it-',
-            dir=os.environ.get('TEST_TMPDIR', None))
+        temp_location = pipeline.options.get_all_options()['temp_location']
```

Comment:
Can we also add staging_location since temp_location is not always required.

Reply:
Changed, thanks!